build.gradle

```diff
@@ -174,6 +174,7 @@ project(':iceberg-bundled-guava') {
 project(':iceberg-api') {
   dependencies {
     compile project(path: ':iceberg-bundled-guava', configuration: 'shadow')
+    compileOnly "com.google.errorprone:error_prone_annotations:2.3.3"
```

Comment:
Required for the FormatString Annotation

Reply:
I think there's still a number of warnings to turn it on. Errorprone is current enforced for certain situations (line length being one notable case), but not all. Though I agree, we should discuss whether or not we want stricter enforcement in place.

As an example, I went in and added a number of `@override` annotations a while back but more have snuck in since then that are missing.

Reply:
I'm not sure what you mean, the palanrir baseline plugin is turning in both spotbugs and errorprone in Gradle. It just doesn't include the libraries as compile time dependencies for modules since they aren't needed unless you use annotations.

I do think once we hit 0 warnings we should switch it to test warnings as errors

Reply:
I am not familiar with `baseline`. I can't find it in `tasks`. How do you run baseline errorprone or spotbugs?

Reply:
They are applied to all the compileJava tasks in the project. So running any compileJava task will cause error-prone to also be run. So as long as the class file is compiled on your run of "build" or classes or any command, it will log warnings and such. There is no separate task to only run error-prone. We have to set another config option if we want to treat warnings as errors. 

For example if you run the `clean testClasses` now you will see things like 

```
/Users/russellspitzer/repos/iceberg/spark/src/main/java/org/apache/iceberg/spark/source/RowDataRewriter.java:92: warning: [ThrowSpecificity] Prefer to declare more specific throws types than Exception and Throwable. When methods are updated to throw new checked exceptions they expect callers to handle failure types explicitly. Throwing broad types defeats the type system. By throwing the most specific types possible we leverage existing compiler functionality to detect unreachable code.
  private List<DataFile> rewriteDataForTask(CombinedScanTask task) throws Exception {
```

Which is an error-prone based warning.

I was wrong about spot bugs, it is not in the baseline plugin. I misread spotless, which is the formatting tool included in the baseline plugin.

Reply:
> I'm not sure what you mean, the palanrir baseline plugin is turning in both spotbugs and errorprone in Gradle. It just doesn't include the libraries as compile time dependencies for modules since they aren't needed unless you use annotations.
> 
> I do think once we hit 0 warnings we should switch it to test warnings as errors

Yes, it's already turned on. What I was referring to and I assumed (potentially incorrectly) that @jzhuge was referring to when saying turning it "on" was to enforce more classes of warnings as errors.

Thanks again for picking up this linting!